src/parser/statement.js

```diff
@@ -771,8 +771,13 @@ pp.parseClassBody = function (node) {
 };
 
 pp.parseClassProperty = function (node) {
+  const noPluginMsg = "You can only use Class Properties when the 'classProperties' plugin is enabled.";
+  if (!node.typeAnnotation && !this.hasPlugin("classProperties")) {
```

Comment:
I'm not super happy about a flow-specific check leaking into the general parser, but I didn't see a straight-forward way to do this otherwise. Only other way I could see was to just have the flow extension for `parseClassProperty` to invoke this with an optional second prop `hadType`.